modules/ztp-image-based-upgrade-prep.adoc

```diff
@@ -0,0 +1,208 @@
+metadata:
+  name: acm-klusterlet
+  namespace: openshift-adp
+  labels:
+    velero.io/storage-location: default
+  annotations:
+    lca.openshift.io/apply-wave: "1"
+spec:
+  backupName:
+    acm-klusterlet
+---
+apiVersion: velero.io/v1
+kind: Restore
+metadata:
+  name: example-app
+  namespace: openshift-adp
+  labels:
+    velero.io/storage-location: default
+  annotations:
+    lca.openshift.io/apply-wave: "3"
+spec:
+  backupName:
+    example-app
+---
+apiVersion: velero.io/v1
+kind: Restore
+metadata:
+  name: localvolume
+  namespace: openshift-adp
+  labels:
+    velero.io/storage-location: default
+  annotations:
+    lca.openshift.io/apply-wave: "2"
+spec:
+  backupName:
+    localvolume
+----
+
+.. Create a `kustomization.yaml` that will append the information to a new `ConfigMap` CR:
++
+[source,yaml]
+----
+configMapGenerator:
+- name: oadp-cm
+  namespace: openshift-adp
+  files:
+  - backup_acm_klusterlet.yaml
+  - backup_localvolume.yaml
+  - backup_app.yaml
+  - restore_acm_klusterlet.yaml
+  - restore_localvolumes.yaml
+  - restore_app.yaml
+generatorOptions:
+  disableNameSuffixHash: true
+----
+
+.. Create the `ConfigMap` CR:
++
+[source,terminal]
+----
+$ kustomize build ./ -o OadpCm.yaml
+----
+
++
+.Example output
+[source,terminal]
+----
+kind: ConfigMap
+metadata:
+  name: oadp-cm-example
+  namespace: openshift-adp
+[...]
+----
+
+.. Apply the `ConfigMap` CR:
++
+[source,terminal]
+----
+$ oc apply -f OadpCm.yaml
+----
+
+. Edit the `ImageBasedUpgrade` CR:
++
+[source,yaml]
+----
+apiVersion: lca.openshift.io/v1alpha1
+kind: ImageBasedUpgrade
+metadata:
+  name: example-upgrade
+spec:
+  stage: Idle
+  seedImageRef:
+    version: 4.14.6 <1>
+    image: <seed_container_image> <2>
+    pullSecretRef: <seed_pull_secret> <3>
+  additionalImages:
+    name: ""
+    namespace: ""
+  autoRollbackOnFailure: {} <4>
+#    disabledForPostRebootConfig: "true" <5>
+#    disabledForUpgradeCompletion: "true" <6>
+#    disabledInitMonitor: "true" <7>
+#    initMonitorTimeoutSeconds: 1800 <8>
+#  extraManifests: <9>
+#  - name: sno-extra-manifests
+#    namespace: openshift-lifecycle-agent
+#  oadpContent: <10>
+#  - name: oadp-cm-example
+#    namespace: openshift-adp
+----
+<1> Specify the platform version on the seed image.
+<2> Specify the repository where the target cluster can pull the seed image from.
+<3> Specify the reference to a secret with credentials to pull container images.
+<4> By default, automatic rollback is enabled throughout the upgrade.
+<5> (Optional) Disables automatic rollback when the reconfiguration of the cluster fails upon the first reboot.
+<6> (Optional) Disables automatic rollback after the {lcao} reports a failed upgrade upon completion.
+<7> (Optional) Disables automatic rollback when the upgrade does not complete after reboot within the time frame specified in the `initMonitorTimeoutSeconds` field.
+<8> Specifies the time frame in seconds. The default value is `1800` seconds or 30 minutes.
```

Comment:
Missing the "optional" tag. If not specified, the default of 1800 seconds is used.